### PR TITLE
Fixed #27699 -- Fixed parse_duration() with a negative number of seconds.

### PR DESCRIPTION
**Pull request by hardik1997**

### CODE REVIEW

File: tests/utils_tests/test_dateparse.py [hunk 1]
```diff
@@ -108,6 +108,11 @@ def test_fractions_of_seconds(self):
 
     def test_negative(self):
         self.assertEqual(parse_duration('-4 15:30'), timedelta(days=-4, minutes=15, seconds=30))
+        self.assertEqual(parse_duration('4 -15:30'), timedelta(days=4, minutes=-15, seconds=30))
+        self.assertEqual(parse_duration('4 15:-30'), timedelta(days=4, minutes=15, seconds=-30))
```
Comment (timgraham) [inline tests/utils_tests/test_dateparse.py]: This doesn't look like correct notation. What about the approach in https://github.com/django/django/pull/7814? I think you need to test this in the context of `DurationField` and SQLite as described on the ticket.

Comment (hardik1997) [inline tests/utils_tests/test_dateparse.py]: Ok. can only first field be negative? like(-4 1:2:3), (-1:2:3), (-1:2) etc?

Comment (hardik1997) [inline tests/utils_tests/test_dateparse.py]: Ok I will submit tests for DurationField also.

**Comment by claudep** [comment]
Closing in favor of #7814. If you have complementary fix or test, please open a new PR.

**Pull request closed without merging**